File: solutions/ruby/hamming/3/hamming.rb
module Hamming
  def self.compute(a, b)
    raise ArgumentError if a.length != b.length

    len = a.length
    diff = 0
    i = 0

    while i < len
      diff += 1 if a.getbyte(i) != b.getbyte(i)
      i += 1
    end

    diff
  end
end

File: solutions/c/hamming/1/hamming.c
#include "hamming.h"
#include <string.h>

int compute(const char *lhs, const char *rhs) {
    if (!lhs || !rhs || (strlen(lhs) != strlen(rhs))) 
        return -1;
        
    int differences = 0;
        
    for (int i = 0; lhs[i] != '\0'; i++) {
        if (lhs[i] != rhs[i]) differences++;
    }
    
    return differences;
}